addons/sale/models/sale.py

```diff
@@ -469,12 +469,15 @@ def _prepare_invoice(self):
             .default_get(['journal_id'])['journal_id'])
         if not journal_id:
             raise UserError(_('Please define an accounting sales journal for this company.'))
-        invoice_vals = {
+        vinvoice = self.env['account.invoice'].new({'partner_id': self.partner_invoice_id.id})
+        # Get partner extra fields
+        vinvoice._onchange_partner_id()
+        invoice_vals = vinvoice._convert_to_write(vinvoice._cache)
+        invoice_vals.update({
```

Comment:
This code move makes sense as in a 'prepare' function!

Reply:
It's already in a prepare function :wink: